core/controllers/ingest_dependencies.py

```diff
@@ -66,6 +66,7 @@ def ingest_dependencies_factory(fund: str, reporting_round: int) -> IngestDepend
                 transform_data=ingest_round_three_data_towns_fund,
                 validation_schema=TF_ROUND_3_VAL_SCHEMA,
                 initial_validation_schema=TF_ROUND_3_INIT_VAL_SCHEMA,
+                messenger=TFMessenger(),
```

Comment:
Is this just for consistency or is it doing something more real than I imagine?

My naive understanding I suppose now is that rounds 3 and 4 are closed and done and dead and that these would never be hit? Am I just totally wrong there?

Reply:
Round 5 uses the same `validation_schema` as Round 4 (as the spreadsheet is the same one), and we sometimes reingest spreadsheets from old rounds in order to tidy up data issues or for data to fit into DB changes - in this case those few spreadsheets you, Gideon and Noel found that have start dates after completion dates will be reingested, so including the validation check for R3 is just to make sure we haven't missed any incorrect dates when those are reingested (this was specified in the JIRA ticket).

The reason for the messenger specifically being added to Round 3 here is that the `user.GenericFailure` validation error that the validation generates needs the messenger to generate the error messages - it's there in r4 and none of the r3 errors used this Class so needed to add the messenger here.

Reply:
Nice, thanks for explaining 👍